lstests/doc/README.md

```diff
@@ -0,0 +1,692 @@
+    No default value.
+extra_configs : list
+    Extra configs to pass to nodeos.
+    e.g. ["plugin=SOME_EXTRA_PLUGIN"]
+    No default value.
+extra_args : str
+    Extra arguments to pass to launcher service.
+    e.g. "--delete-all-blocks"
+    No default value.
+dont_newacco : bool
+    Do not create accounts in launch.
+    Default is False (will create producer accounts).
+dont_setprod : bool
+    Do not set producers in launch.
+    Default is False (will set producers).
+http_retry : int
+    Max number of retries in HTTP connection.
+    Default is 100.
+http_sleep : float
+    Sleep time (in seconds) between HTTP connection retries.
+    Default is 0.25.
+verify_async : bool
+    Verify transactions asynchronously.
+    Start a separate thread for transaction verfication. Do not wait
+    for a transaction to be verified before making next transaction.
+    Default is False (will wait for verification result).
+verify_retry : int
+    Max number of retries in transaction verfication.
+    Default is 100.
+verify_sleep : float
+    Sleep time (in seconds) between verfication retries.
+    Default is 0.25.
+sync_retry : int
+    Max number of retries in checking if nodes are in sync.
+    Default is 100.
+sync_sleep : float
+    Sleep time (in seconds) between check-sync retries.
+    Default is 0.25.
+```
+
+For `extra_configs`, the user may refer to the help text that `programs/nodeos/nodeos -h` offers.
+
+For `extra_args`, the user may refer to the help text that `programs/launcher-service/launcher-service -h` offers.
+
+For example, by launcher service's help text
+
+```bash
+--genesis-json arg (=genesis.json)    path of genesis file
+```
+
+the user may create a cluster with a specific genesis file.
+
+```python
+clus = Cluster(service=service, extra_args="--genesis-json=gene.json")
+```
+
+This specifies `gene.json` as the genesis file for the cluster, which overrides the service-level setting.
+
+## Testing
+
+The `Cluster` class provides a list of methods, such as `set_contract()` and `push_actions()`, which serve as building blocks for a test script to build up the testing scenario.
+
+### Mechanism
+
+Behind the scenes, a `Cluster` object communicates with the launcher service via HTTP connection in a request-and-response model.
+
+Almost all the methods finally passes through or are dependent on the `call()`  method in the `Cluster` class. The standard steps to make a call include
+
+```
+  Steps to take to make a call
+  ----------------------------
+  1. print header
+  2. establish HTTP connection
+  3. log URL and request of connection
+  4. retry connection if response not ok
+  5. log response
+  6. verify transaction
+  7. return the result as a Connection object
+```
+
+The exact actions to take are fully customizable. Refer to `**call_kwargs` for the full control.
+
+### Main API
+
+After the node cluster is successfully launched, all the test actions can be performed on the cluster, with main API listed below.
```

Comment:
"on the cluster, with main API listed below." -> "on the cluster. The main API is listed below."

Reply:
Updated. Thank you.